File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        Queue<String> q1 = new LinkedList<>();
        Queue<String> q2 = new LinkedList<>();

        for (String card : cards1) {
            q1.offer(card);
        }
        for (String card : cards2) {
            q2.offer(card);
        }

        for (String s : goal) {
            if (!q1.isEmpty() && q1.peek().equals(s)) q1.poll();
            else if (!q2.isEmpty() && q2.peek().equals(s)) q2.poll();
            else return "No";
        }
        return "Yes";
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.kt
import java.util.*

class Solution {fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        var answer: String = "Yes"
        val q1 = LinkedList<String>()
        val q2 = LinkedList<String>()

        cards1.forEach{ q1.offer(it) }
        cards2.forEach{ q2.offer(it) }

        for (s in goal){
            if (q1.isNotEmpty() && q1.peek().equals(s)) q1.poll()
            else if (q2.isNotEmpty() && q2.peek().equals(s)) q2.poll()
            else{
                answer = "No"
                break
            }
        }

        return answer
    }

}